File: Solutions/[11]_2.cpp
#include <bits/stdc++.h>

using namespace std;

int n, result;
// C++에서는 기본적으로 우선순위 큐가 최대 힙(Max Heap)
priority_queue<int> pq;

int main(void) {
	cin >> n;
	
	for (int i = 0; i < n; i++) {
		int x;
		cin >> x;
		pq.push(-x);
	}
	
	while (pq.size() != 1) {
		int one = pq.top();
		pq.pop();
		int two = pq.top();
		pq.pop();
		int sum_value = one + two;
		result += sum_value;
		pq.push(sum_value);
	}
	
	cout << -result;
}

File: Solutions/[11]_2.java
import java.util.PriorityQueue;
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		int n = sc.nextInt();
		int result = 0;
		// 자바(Java)에서는 기본적으로 우선순위 큐가 최소 힙(Min Heap)
		PriorityQueue<Integer> pq = new PriorityQueue<Integer>();
		
		for (int i = 0; i < n; i++) {
			int x = sc.nextInt();
			pq.add(x);
		}
		
		while (pq.size() != 1) {
			int one = pq.poll();
			int two = pq.poll();
			int sum_value = one + two;
			result += sum_value;
			pq.add(sum_value);
		}
		
		System.out.println(result);
	}

}
